### PR TITLE
fix(assertions): honor inverse (not-) for levenshtein assertion

### PR DESCRIPTION
**Pull request by i-anubhav-anand**
## Summary

`handleLevenshtein` never read the `inverse` flag, so the `not-levenshtein`
assertion behaved **identically** to `levenshtein`: it passed whenever the edit
distance was within the threshold, regardless of the `not-` negation.

The framework derives `inverse` from the `not-` prefix (`assertions/index.ts` →
`isAssertionInverse`) and passes it to every handler; there is no generic
post-handler inversion, so each handler must honor it. `levenshtein` now computes
`pass` via `levDistance <= threshold !== inverse`, matching the sibling threshold
assertions (`bleu`/`rouge`/`gleu`/`cost`/`latency`/`perplexity`), and branches its
failure `reason` on `inverse`.

> Scope note: the sibling `cost` / `latency` / `perplexity` / `perplexity-score`
> fixes landed separately in #9725. This PR is now scoped to the `levenshtein`
> assertion only.

### Before / After

```yaml
assert:
  - type: not-levenshtein
    value: hello
    threshold: 2
# output: "hello"  (distance 0, within threshold)
```

- **Before:** `inverse` ignored → `pass = 0 <= 2 = true`. The `not-` assertion
  passes even though the output *is* within the threshold (the opposite of intent).
- **After:** `pass = (0 <= 2) !== true = false`. Correct.

## Change Type

- [x] Bug fix (non-breaking change which fixes an issue)

## Testing

- `npx vitest run test/assertions/levenshtein.test.ts test/assertions/runAssertion.test.ts`
  — green. Levenshtein regression tests: inverse within-threshold → fail,
  over-threshold → pass, **exact-boundary** (`distance === threshold`) → fail, and
  the **default-threshold** inverse path. Plus `runAssertion` integration coverage
  for the `not-` threshold variants that complements the dedicated unit tests in
  #9725. Each new test fails against the pre-fix code.
- Full `test/assertions/` suite (1203 tests), `npm run tsc`, and Biome
  lint/format all clean. `levenshtein.ts` is at 100% line coverage.
- End-to-end via the built CLI (`promptfoo eval`): `not-levenshtein` with a
  distance-0 output correctly **fails** (`Levenshtein distance 0 is less than or
  equal to threshold 2`).

### CODE REVIEW

File: src/assertions/perplexity.ts [hunk 1]
```diff
@@ -30,15 +39,18 @@ export function handlePerplexityScore({ logProbs, assertion }: AssertionParams):
...
   const perplexityNorm = 1 / (1 + perplexity);
 
-  const pass = assertion.threshold === undefined ? true : perplexityNorm >= assertion.threshold;
+  // With no threshold there is nothing to compare against, so the assertion always
+  // passes — including its `not-` variant. Otherwise honor the `not-` negation.
+  const pass =
+    assertion.threshold === undefined ? true : perplexityNorm >= assertion.threshold !== inverse;
   return {
     pass,
     score: perplexityNorm,
```
[the remark this answers is not in the document]
Comment (mldangelo) [reply src/assertions/perplexity.ts]: Addressed in 1e6ed3f0e2 — `handlePerplexityScore` now returns `score: inverse ? 1 - perplexityNorm : perplexityNorm`, mirroring `bleu.ts`/`gleu.ts`/`matchers/similarity.ts`. Added tests pinning both the inverted (`not-`) and forward scores. Thanks for the catch — an independent red-team pass flagged the same aggregation issue.

**Review by mldangelo — APPROVED** [review]
Thanks so much for this fix, @i-anubhav-anand — and for the thorough write-up. 🙏

`not-levenshtein` silently behaving like `levenshtein` was a real correctness bug, and your before/after analysis (and the sibling precedent with `similar`) made it easy to verify. I scoped this PR down to the `levenshtein` change since your follow-up #9725 already landed the `cost` / `latency` / `perplexity` siblings on `main`, and added a few extra regression tests (exact-boundary `distance === threshold`, the default-threshold inverse path, and integration coverage through the `runAssertion` dispatch).

Verified locally and in CI: the full assertions suite passes, types/lint are clean, and an end-to-end `promptfoo eval` confirms `not-levenshtein` now correctly fails on a within-threshold output. Really appreciate you splitting the work cleanly and following up on the other handlers — great contribution. Approving and merging.
